Adds VideoDecoder (#8984)
mdn/content: files/en-us/web/api/videodecoder/index.md

---
title: VideoDecoder
slug: Web/API/VideoDecoder
tags:
  - API
  - Interface
  - Reference
  - VideoDecoder
browser-compat: api.VideoDecoder
---
{{securecontext_header}}{{DefaultAPISidebar("WebCodecs API")}}

> The **`VideoDecoder`** interface of the {{domxref('WebCodecs API','','',' ')}} decodes chunks of video.

Reviewer: ```suggestion
The **`VideoDecoder`** interface of the {{domxref('WebCodecs API','','','true')}} decodes chunks of video.
```

Reviewer: This makes the title not monospaced.

## Constructor

- {{domxref("VideoDecoder.VideoDecoder()")}}
  - : Creates a new `VideoDecoder` object.

## Properties

- {{domxref("VideoDecoder.decodeQueueSize")}}{{ReadOnlyInline}}
  - : An integer representing the number of decode queue requests.
- {{domxref("VideoDecoder.state")}}{{ReadOnlyInline}}

>   - : Represents the state of the underlying codec and whether it is configured for decoding.

Reviewer: ```suggestion
  - : Indicates whether the underlying codec is configured for decoding.
```
